src/sys/unix/epoll.rs

```diff
@@ -88,9 +88,9 @@ impl Selector {
     }
 
     /// Register event interests for the given IO handle with the OS
-    pub fn register(&self, fd: RawFd, token: Token, interests: Ready, opts: PollOpt) -> io::Result<()> {
+    pub fn register(&self, fd: RawFd, token: Token, interests: Interests, opts: PollOpt) -> io::Result<()> {
         let mut info = libc::epoll_event {
-            events: ioevent_to_epoll(interests, opts),
+            events: ioevent_to_epoll(Ready::from_usize(interests.as_usize()), opts),
```

Comment:
Why not just use `Intersts` in `ioevent_to_epoll`?

Reply:
Because this:
```rust
    pub fn push_event(&mut self, event: Event) {
        self.events.push(libc::epoll_event {
            events: ioevent_to_epoll(event.readiness(), PollOpt::empty()),
            u64: usize::from(event.token()) as u64,
        });
    }
```

`Event::readiness` return `Ready` and `ioevent_to_epoll` accept it

Reply:
Maybe we create two separate fns?

Reply:
`ready_to_epoll` and `interests_to_epoll` ?

Reply:
Maybe just use `Interests::to_ready` to convert and use one `ioevent_to_epoll`?